gui/default/syncthing/core/uniqueFolderDirective.js

```diff
@@ -4,7 +4,7 @@ angular.module('syncthing.core')
             require: 'ngModel',
             link: function (scope, elm, attrs, ctrl) {
                 ctrl.$parsers.unshift(function (viewValue) {
-                    if (scope.currentFolder._editing != "add") {
+                    if (!scope.editingFolderNew()) {
```

Comment:
This branch should also match the "new-pending" case, because just as with existing folders the ID entry will be readonly.  Validating without being able to change it is pointless.

Reply:
That's exactly the downside of these utility functions: It's not immediately obvious what they do. This one here already does match `new` or `new-pending`, as both are variants of adding a new folder.

Reply:
Yes, I meant the negation should match "new-pending".  In effect, this should be

    if (scope.currentFolder._editing != "new") {

Reply:
Right, there's already code checking for existing folders with that ID to present the user with the option to share the folder instead of adding it, so at this point we now it's unique -> adjusted.